// react-native-magnus/lib/commonjs/theme/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ThemeContext", {
  enumerable: true,
  get: function () {
    return _theme.ThemeContext;
  }
});
Object.defineProperty(exports, "ThemeProvider", {
  enumerable: true,
  get: function () {
    return _theme2.ThemeProvider;
  }
});
Object.defineProperty(exports, "ThemeProviderProps", {
  enumerable: true,
  get: function () {
    return _theme2.ThemeProviderProps;
  }
});
Object.defineProperty(exports, "ThemeType", {
  enumerable: true,
  get: function () {
    return _type.ThemeType;
  }
});
Object.defineProperty(exports, "getThemeProperty", {
  enumerable: true,
  get: function () {
    return _theme3.getThemeProperty;
  }
});
Object.defineProperty(exports, "useTheme", {
  enumerable: true,
  get: function () {
    return _theme4.useTheme;
  }
});
Object.defineProperty(exports, "withTheme", {
  enumerable: true,
  get: function () {
    return _withTheme.withTheme;
  }
});

var _withTheme = require("./withTheme");

var _theme = require("./theme.context");

var _theme2 = require("./theme.provider");

var _theme3 = require("./theme.service");

var _theme4 = require("./theme.hook");

var _type = require("./type");
