// react-native-magnus/lib/module/ui/checkbox/checkbox.component.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import { useState, useEffect } from 'react';
import { View as RNView, Pressable as RNButton } from 'react-native';
import { getStyle } from './checkbox.style';
import { isFunction } from '../../utilities';
import { getIcon } from './checkbox.service';
import { CheckboxGroup } from './group.component';
import { useDefaultProps } from '../../utilities/useDefaultProps';
import { useTheme } from '../../theme';

const Checkbox = incomingProps => {
  const props = useDefaultProps('Checkbox', incomingProps, {
    defaultChecked: false,
    activeColor: 'blue600',
    inactiveColor: 'gray500',
    highlightBg: 'gray300',
    bg: 'transparent',
    p: 'none',
    color: 'white',
    rounded: 'md',
    loading: false,
    disabled: false,
    loaderSize: '2xl',
    loaderColor: 'blue600',
    block: false,
    position: 'relative',
    shadowColor: 'gray800',
    shadow: 0,
    fontSize: '5xl',
    borderless: true,
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'flex-start',
    onPress: () => {},
    flexDir: 'row'
  });
  const {
    m,
    mt,
    mr,
    mb,
    ml,
    ms,
    p,
    pr,
    pt,
    pb,
    pl,
    h,
    w,
    bg,
    minW,
    minH,
    suffix,
    style,
    fontSize,
    fontWeight,
    prefix,
    rounded,
    roundedTop,
    roundedRight,
    roundedBottom,
    roundedLeft,
    color: colorProp,
    loading,
    disabled,
    loaderColor,
    loaderSize,
    children,
    borderColor,
    borderBottomColor,
    borderLeftColor,
    borderTopColor,
    borderRightColor,
    borderWidth,
    borderLeftWidth,
    borderRightWidth,
    borderBottomWidth,
    borderTopWidth,
    borderEndWidth,
    shadow,
    borderless,
    shadowColor,
    block,
    alignSelf,
    activeIcon,
    inactiveIcon,
    onChange,
    value,
    activeColor,
    inactiveColor,
    defaultChecked,
    checked: checkedProp,
    onChecked,
    onPress: onPressProp,
    ...rest
  } = props;
  const {
    theme
  } = useTheme();
  const [checked, setChecked] = useState(('checked' in props ? checkedProp : defaultChecked) ?? false);
  const [focussed, setFocussed] = useState(false);
  const computedStyle = getStyle(theme, props, {
    focussed
  });
  useEffect(() => {
    if ('checked' in props) {
      setChecked(props.checked ?? false);
    }
  }, [props]);
  /**
   * on press checkbox
   */

  const onPress = event => {
    if (disabled) {
      return;
    }

    setChecked(!checked);

    if (isFunction(onChecked)) {
      onChecked(!checked);
    }

    if (isFunction(onPressProp)) {
      onPressProp(event);
    } // if onChange prop is a valid function, call it


    if (isFunction(onChange)) {
      onChange(value);
    }
  };
  /**
   * sets focussed to true
   *
   * @param event
   */


  const onPressIn = () => {
    setFocussed(true);
  };
  /**
   * sets focussed to true
   *
   * @param event
   */


  const onPressOut = () => {
    setFocussed(false);
  };

  const renderChildren = () => {
    if (isFunction(children)) {
      return children({
        focussed,
        disabled: disabled ?? false,
        checked,
        loading
      });
    }

    return /*#__PURE__*/React.createElement(React.Fragment, null, prefix, /*#__PURE__*/React.createElement(RNView, null, /*#__PURE__*/React.createElement(RNView, {
      style: computedStyle.highlightContainer
    }), /*#__PURE__*/React.createElement(RNView, {
      style: computedStyle.icon
    }, icon)), children, suffix);
  };

  const icon = getIcon(theme, props, checked);
  return /*#__PURE__*/React.createElement(RNButton, _extends({}, rest, {
    style: computedStyle.button,
    onPress: disabled || loading ? undefined : onPress,
    onPressIn: disabled || loading ? undefined : onPressIn,
    onPressOut: disabled || loading ? undefined : onPressOut
  }), /*#__PURE__*/React.createElement(RNView, {
    style: computedStyle.container
  }, renderChildren()));
}; // Checkbox.defaultProps = {
//   defaultChecked: false,
//   activeColor: 'blue600',
//   inactiveColor: 'gray500',
//   highlightBg: 'gray300',
//   bg: 'transparent',
//   p: 'none',
//   color: 'white',
//   rounded: 'md',
//   loading: false,
//   disabled: false,
//   loaderSize: '2xl',
//   loaderColor: 'blue600',
//   block: false,
//   position: 'relative',
//   shadowColor: 'gray800',
//   shadow: 0,
//   fontSize: '5xl',
//   borderless: true,
//   alignItems: 'center',
//   justifyContent: 'center',
//   alignSelf: 'flex-start',
//   onPress: () => {},
//   flexDir: 'row',
// };


Checkbox.Group = CheckboxGroup;
export { Checkbox };
