// react-native-magnus/lib/module/ui/input/input.component.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import color from 'color';
import { useState } from 'react';
import { View as RNView, TextInput as RNTextInput, ActivityIndicator as RNActivityIndicator } from 'react-native';
import { getStyle } from './input.style';
import { useTheme } from '../../theme';
import { getThemeProperty, getThemeColor } from '../../theme/theme.service';
import { useDefaultProps } from '../../utilities/useDefaultProps';
const Input = /*#__PURE__*/React.forwardRef((incomingProps, ref) => {
  const props = useDefaultProps('Input', incomingProps, {
    px: 'lg',
    py: 'lg',
    fontSize: 'md',
    borderWidth: 1,
    bg: 'white',
    borderColor: 'gray400',
    rounded: 'md',
    loading: false,
    color: 'gray800',
    shadow: 0,
    shadowColor: 'gray500',
    loaderSize: '3xl',
    loaderColor: 'blue700'
  });
  const {
    h,
    w,
    m,
    mt,
    mr,
    mb,
    ml,
    ms,
    p,
    pr,
    pt,
    pb,
    pl,
    bg,
    minH,
    minW,
    color: colorProp,
    suffix,
    prefix,
    style,
    onBlur,
    loading,
    onFocus,
    rounded,
    roundedTop,
    roundedRight,
    roundedBottom,
    roundedLeft,
    children,
    loaderSize,
    fontSize,
    borderColor,
    borderBottomColor,
    borderLeftColor,
    borderTopColor,
    borderRightColor,
    borderWidth,
    borderLeftWidth,
    borderRightWidth,
    borderBottomWidth,
    borderTopWidth,
    borderEndWidth,
    loaderColor,
    focusBorderColor,
    shadow,
    flex,
    shadowColor,
    zIndex,
    opacity,
    placeholderTextColor,
    selectionColor,
    ...rest
  } = props;
  const {
    theme
  } = useTheme();
  const [isFocussed, setIsFocussed] = useState(false);
  const computedStyle = getStyle(theme, props, {
    isFocussed
  });
  const placeholderColor = placeholderTextColor ? typeof placeholderTextColor === 'string' ? color(getThemeColor(theme.colors, placeholderTextColor)).alpha(0.4).rgb().string() : placeholderTextColor : color(getThemeColor(theme.colors, colorProp)).alpha(0.4).rgb().string();
  /**
   * on focus input
   *
   * @param e
   */

  const onFocusInput = e => {
    setIsFocussed(true);

    if (onFocus) {
      onFocus(e);
    }
  };
  /**
   * on blur input
   *
   * @param e
   */


  const onBlurInput = e => {
    setIsFocussed(false);

    if (onBlur) {
      onBlur(e);
    }
  };

  return /*#__PURE__*/React.createElement(RNView, {
    style: computedStyle.container
  }, prefix && /*#__PURE__*/React.createElement(RNView, {
    style: computedStyle.prefix
  }, prefix), /*#__PURE__*/React.createElement(RNTextInput, _extends({
    ref: ref,
    onFocus: onFocusInput,
    onBlur: onBlurInput,
    selectionColor: typeof props.selectionColor === 'string' ? getThemeColor(theme.colors, props.selectionColor) : props.selectionColor
  }, rest, {
    style: computedStyle.input,
    placeholderTextColor: placeholderColor
  })), !loading && suffix && /*#__PURE__*/React.createElement(RNView, {
    style: computedStyle.suffix
  }, suffix), loading && /*#__PURE__*/React.createElement(RNView, {
    style: computedStyle.suffix
  }, /*#__PURE__*/React.createElement(RNActivityIndicator, {
    size: getThemeProperty(theme.fontSize, loaderSize),
    color: getThemeColor(theme.colors, loaderColor)
  })));
}); // Input.defaultProps = {
//   px: 'xl',
//   py: 'lg',
//   fontSize: 'lg',
//   borderWidth: 0,
//   bg: 'white',
//   borderColor: 'gray400',
//   rounded: 'md',
//   loading: false,
//   color: 'gray800',
//   shadow: 0,
//   shadowColor: 'gray500',
//   loaderSize: '3xl',
//   loaderColor: 'blue700',
// };

export { Input };
