// react-native-magnus/lib/commonjs/ui/header/header.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getStyle = void 0;

var _reactNative = require("react-native");

/**
 * computed styles
 *
 * @param theme
 * @param props
 */
const getStyle = (_theme, props) => {
  const computedStyle = {}; // merging style props to computed style

  if (props.style) {
    computedStyle.container = { ...computedStyle.container,
      // @ts-ignore
      ...props.style
    };
  }

  computedStyle.center = {};

  if (props.suffix) {
    computedStyle.suffix = {
      zIndex: 1,
      flexDirection: 'row',
      alignItems: 'center'
    };
  }

  if (props.prefix) {
    computedStyle.prefix = {
      zIndex: 1,
      flexDirection: 'row',
      alignItems: 'center'
    };
  }

  if (props.alignment === 'center') {
    computedStyle.center = { ...computedStyle.center,
      justifyContent: 'center',
      alignItems: 'center',
      flex: 1
    };
    computedStyle.container = { ...computedStyle.container,
      justifyContent: 'space-between'
    };

    if (props.suffix || props.prefix) {
      computedStyle.center = { ...computedStyle.center,
        ..._reactNative.StyleSheet.absoluteFillObject
      };
      computedStyle.suffix = { ...computedStyle.suffix,
        flex: 1,
        justifyContent: 'flex-end'
      };
    }
  } else {
    computedStyle.center = {
      flex: 1
    };
  }

  return _reactNative.StyleSheet.create(computedStyle);
};

exports.getStyle = getStyle;
