// react-native-magnus/lib/commonjs/ui/badge/badge.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getStyle = void 0;

var _reactNative = require("react-native");

var _theme = require("../../theme/theme.service");

/**
 * computed style
 *
 * @param theme
 * @param props
 */
const getStyle = (theme, props) => {
  const computedStyle = {};
  computedStyle.container = {
    alignSelf: props.alignSelf,
    minHeight: typeof props.count === 'undefined' ? 10 : 30,
    minWidth: typeof props.count === 'undefined' ? 10 : 30
  };
  computedStyle.div = {
    zIndex: 1,
    height: props.h,
    width: props.w,
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'flex-start',
    opacity: props.opacity,
    minHeight: typeof props.children === 'string' ? 30 : 10,
    backgroundColor: (0, _theme.getThemeColor)(theme.colors, props.bg),
    minWidth: typeof props.children === 'string' ? 30 : 10,
    ...(0, _theme.createPositionStyle)(props),
    ...(0, _theme.createBorderWidthStyles)(props),
    ...(0, _theme.createShadowStyles)(props, theme),
    ...(0, _theme.createSpacingStyles)(props, theme.spacing),
    ...(0, _theme.createBorderColorStyles)(props, theme.colors),
    ...(0, _theme.createBorderRadiusStyles)(props, theme.borderRadius),
    position: typeof props.children !== 'string' ? 'absolute' : props.position
  };
  computedStyle.text = {
    paddingHorizontal: 10
  }; // merging style props to computed style

  if (props.style) {
    computedStyle.div = { ...computedStyle.div,
      // @ts-ignore
      ...props.style
    };
  }

  return _reactNative.StyleSheet.create(computedStyle);
};

exports.getStyle = getStyle;
