// react-native-magnus/lib/commonjs/ui/collapse/collapse.header.component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CollapseHeader = void 0;

var React = _interopRequireWildcard(require("react"));

var _icon = require("../icon/icon.component");

var _button = require("../button/button.component");

var _div = require("../div/div.component");

var _useDefaultProps = require("../../utilities/useDefaultProps");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const CollapseHeader = incomingProps => {
  const props = (0, _useDefaultProps.useDefaultProps)('CollapseHeader', incomingProps, {
    block: true,
    roundedBottom: 'none',
    bg: 'blue600',
    color: 'white',
    py: 'xl',
    px: 'none',
    justifyContent: 'flex-start',
    prefix: /*#__PURE__*/React.createElement(_div.Div, {
      px: "lg"
    }),
    suffix: /*#__PURE__*/React.createElement(_icon.Icon, {
      px: "xl",
      name: "chevron-down",
      fontFamily: "Feather",
      position: "absolute",
      right: 0,
      color: "white"
    }),
    activeSuffix: /*#__PURE__*/React.createElement(_icon.Icon, {
      px: "xl",
      name: "chevron-up",
      fontFamily: "Feather",
      position: "absolute",
      right: 0,
      color: "white"
    })
  });
  const {
    suffix,
    activeSuffix,
    prefix,
    activePrefix,
    active
  } = props;
  return /*#__PURE__*/React.createElement(_button.Button, _extends({}, props, {
    prefix: active ? activePrefix ?? prefix : prefix,
    suffix: active ? activeSuffix ?? suffix : suffix
  }));
}; // CollapseHeader.defaultProps = {
//   block: true,
//   roundedBottom: 'none',
//   bg: 'blue600',
//   color: 'white',
//   py: 'xl',
//   px: 'none',
//   justifyContent: 'flex-start',
//   prefix: <Div px="lg" />,
//   suffix: (
//     <Icon
//       px="xl"
//       name="chevron-down"
//       fontFamily="Feather"
//       position="absolute"
//       right={0}
//       color="white"
//     />
//   ),
//   activeSuffix: (
//     <Icon
//       px="xl"
//       name="chevron-up"
//       fontFamily="Feather"
//       position="absolute"
//       right={0}
//       color="white"
//     />
//   ),
// };


exports.CollapseHeader = CollapseHeader;
