// react-native-magnus/lib/module/ui/skeleton/skeleton.component.js
import * as React from 'react';
import * as Animatable from 'react-native-animatable';
import { useDefaultProps } from '../../utilities/useDefaultProps';
import { Div } from '../div/div.component';

const Skeleton = incomingProps => {
  const props = useDefaultProps('Skeleton', incomingProps, {
    bg: 'gray400',
    h: 15,
    w: '100%',
    rounded: 'lg',
    duration: 1000
  });
  const {
    duration,
    ...rest
  } = props;
  Animatable.initializeRegistryWithDefinitions({
    fade: {
      0: {
        opacity: 1
      },
      0.5: {
        opacity: 0.3
      },
      1: {
        opacity: 1
      }
    }
  });
  return /*#__PURE__*/React.createElement(Animatable.View, {
    animation: "fade",
    duration: duration,
    easing: "linear",
    iterationCount: "infinite"
  }, /*#__PURE__*/React.createElement(Div, rest));
};

export const Circle = incomingProps => {
  const props = useDefaultProps('SkeletonCircle', incomingProps, {
    bg: 'gray400',
    h: 15,
    w: 15,
    rounded: 'circle'
  });
  return /*#__PURE__*/React.createElement(Skeleton, props);
};
Skeleton.defaultProps = {
  bg: 'gray400',
  h: 15,
  w: '100%',
  rounded: 'lg',
  duration: 1000
};
Circle.defaultProps = {
  bg: 'gray400',
  h: 15,
  w: 15,
  rounded: 'circle'
};
Skeleton.Box = Skeleton;
Skeleton.Circle = Circle;
export { Skeleton };
