// react-native-magnus/lib/commonjs/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.zIndexProps = exports.variantProps = exports.textProps = exports.spacingProps = exports.shadowProps = exports.roundedProps = exports.preffixSuffixProps = exports.positionProps = exports.overlayProps = exports.overflowProps = exports.opacityProps = exports.loadingProps = exports.inputProps = exports.flexProps = exports.disabledProps = exports.dimensionProps = exports.buttonProps = exports.borderProps = exports.backgroundProps = void 0;
const borderProps = ['borderColor', 'borderTopColor', 'borderRightColor', 'borderBottomColor', 'borderLeftColor', 'borderWidth', 'borderTopWidth', 'borderRightWidth', 'borderLeftWidth', 'borderBottomWidth', 'borderStartWidth', 'borderEndWidth', 'borderStyle'];
exports.borderProps = borderProps;
const spacingProps = ['m', 'mt', 'mr', 'mb', 'ml', 'mx', 'my', 'ms', 'p', 'pt', 'pr', 'pb', 'pl', 'px', 'py', 'ps'];
exports.spacingProps = spacingProps;
const roundedProps = ['rounded', 'roundedTopLeft', 'roundedTopRight', 'roundedBottomLeft', 'roundedBottomRight', 'roundedTop', 'roundedLeft', 'roundedRight', 'roundedBottom'];
exports.roundedProps = roundedProps;
const shadowProps = ['shadow', 'shadowColor'];
exports.shadowProps = shadowProps;
const dimensionProps = ['minH', 'minW', 'maxH', 'maxW', 'h', 'w'];
exports.dimensionProps = dimensionProps;
const flexProps = ['row', 'flex', 'flexDir', 'flexWrap', 'justifyContent', 'alignSelf', 'alignItems'];
exports.flexProps = flexProps;
const positionProps = ['position', 'top', 'right', 'bottom', 'left'];
exports.positionProps = positionProps;
const backgroundProps = ['bg', 'bgImg', 'bgMode'];
exports.backgroundProps = backgroundProps;
const textProps = ['color', 'fontSize', 'textDecorLine', 'textDecorStyle', 'fontStyle', 'textDecorColor', 'fontWeight', 'fontFamily', 'lineHeight', 'textAlign', 'textTransform', 'letterSpacing', 'textAlignVertical'];
exports.textProps = textProps;
const opacityProps = ['opacity'];
exports.opacityProps = opacityProps;
const overflowProps = ['overflow'];
exports.overflowProps = overflowProps;
const zIndexProps = ['zIndex'];
exports.zIndexProps = zIndexProps;
const loadingProps = ['loading', 'loaderSize', 'loaderColor'];
exports.loadingProps = loadingProps;
const preffixSuffixProps = ['preffix', 'suffix'];
exports.preffixSuffixProps = preffixSuffixProps;
const inputProps = ['focusBorderColor'];
exports.inputProps = inputProps;
const disabledProps = ['disabled'];
exports.disabledProps = disabledProps;
const buttonProps = ['underlayColor', 'block', 'borderless', 'rippleColor', 'ripple'];
exports.buttonProps = buttonProps;
const overlayProps = ['overlayColor', 'overlayOpacity'];
exports.overlayProps = overlayProps;
const variantProps = ['variant'];
exports.variantProps = variantProps;
