// react-native-magnus/lib/commonjs/ui/drawer/drawer.component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Drawer = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactNativeModal = _interopRequireDefault(require("react-native-modal"));

var _reactNative = require("react-native");

var _drawer = require("./drawer.style");

var _theme = require("../../theme");

var _theme2 = require("../../theme/theme.service");

var _useDefaultProps = require("../../utilities/useDefaultProps");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Drawer = /*#__PURE__*/React.forwardRef((incomingProps, ref) => {
  const props = (0, _useDefaultProps.useDefaultProps)('Drawer', incomingProps, {
    drawerPercentage: 70,
    animationTime: 300,
    bg: 'white',
    backdropColor: 'black',
    direction: 'left',
    rounded: 'none',
    backdropTransitionOutTiming: 0
  });
  const {
    direction,
    drawerPercentage,
    animationTime,
    backdropColor,
    animationInTiming,
    animationIn,
    animationOut,
    style,
    onBackdropPress,
    isVisible,
    animationOutTiming,
    swipeDirection,
    onSwipeComplete,
    children,
    ...rest
  } = props;
  const {
    theme
  } = (0, _theme.useTheme)();
  const computedStyle = (0, _drawer.getStyle)(theme, props);
  const [visible, setVisible] = (0, React.useState)(isVisible);
  (0, React.useEffect)(() => {
    if ('isVisible' in props) {
      setVisible(props.isVisible ?? false);
    }
  }, [props, visible]);
  /**
   * exposing functions to parent
   */

  (0, React.useImperativeHandle)(ref, () => ({
    open() {
      setVisible(true);
    },

    close() {
      setVisible(false);
    }

  }));
  return /*#__PURE__*/React.createElement(_reactNativeModal.default, _extends({
    isVisible: visible,
    onSwipeComplete: () => setVisible(false),
    swipeDirection: direction === 'left' ? 'left' : 'right',
    backdropColor: (0, _theme2.getThemeColor)(theme.colors, backdropColor),
    animationInTiming: animationInTiming ? animationInTiming : animationTime,
    animationOutTiming: animationOutTiming ? animationOutTiming : animationTime,
    animationIn: direction === 'left' ? 'slideInLeft' : 'slideInRight',
    animationOut: direction === 'left' ? 'slideOutLeft' : 'slideOutRight',
    onBackdropPress: 'onBackdropPress' in props ? onBackdropPress : () => setVisible(false),
    style: computedStyle.drawer
  }, rest), /*#__PURE__*/React.createElement(_reactNative.View, {
    style: computedStyle.container
  }, /*#__PURE__*/React.createElement(_reactNative.SafeAreaView, {
    style: computedStyle.safeView
  }, children)));
}); // Drawer.defaultProps = {
//   drawerPercentage: 70,
//   animationTime: 300,
//   bg: 'white',
//   backdropColor: 'black',
//   direction: 'left',
//   rounded: 'none',
//   backdropTransitionOutTiming: 0,
// };

exports.Drawer = Drawer;
