// react-native-magnus/lib/commonjs/ui/drawer/drawer.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getStyle = void 0;

var _reactNative = require("react-native");

var _theme = require("../../theme/theme.service");

const SCREEN_WIDTH = _reactNative.Dimensions.get('window').width;

const SCREEN_HEIGHT = _reactNative.Dimensions.get('window').height;
/**
 * computed style
 *
 * @param theme
 * @param props
 */


const getStyle = (theme, props) => {
  const computedStyle = {};
  computedStyle.drawer = {
    margin: 0
  };
  computedStyle.container = {
    position: 'absolute',
    height: SCREEN_HEIGHT,
    // @ts-ignore
    width: SCREEN_WIDTH * (props.drawerPercentage / 100),
    zIndex: 0,
    margin: 0,
    backgroundColor: (0, _theme.getThemeColor)(theme.colors, props.bg),
    ...(0, _theme.createBorderWidthStyles)(props),
    ...(0, _theme.createBorderColorStyles)(props, theme.colors),
    ...(0, _theme.createBorderRadiusStyles)(props, theme.borderRadius)
  };
  computedStyle.safeView = {
    flex: 1
  };

  if (props.direction === 'right') {
    computedStyle.container = { ...computedStyle.container,
      right: 0
    };
  } else {
    computedStyle.container = { ...computedStyle.container,
      left: 0
    };
  }

  return _reactNative.StyleSheet.create(computedStyle);
};

exports.getStyle = getStyle;
