// react-native-magnus/lib/module/ui/collapse/collapse.header.component.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import { Icon } from '../icon/icon.component';
import { Button } from '../button/button.component';
import { Div } from '../div/div.component';
import { useDefaultProps } from '../../utilities/useDefaultProps';

const CollapseHeader = incomingProps => {
  const props = useDefaultProps('CollapseHeader', incomingProps, {
    block: true,
    roundedBottom: 'none',
    bg: 'blue600',
    color: 'white',
    py: 'xl',
    px: 'none',
    justifyContent: 'flex-start',
    prefix: /*#__PURE__*/React.createElement(Div, {
      px: "lg"
    }),
    suffix: /*#__PURE__*/React.createElement(Icon, {
      px: "xl",
      name: "chevron-down",
      fontFamily: "Feather",
      position: "absolute",
      right: 0,
      color: "white"
    }),
    activeSuffix: /*#__PURE__*/React.createElement(Icon, {
      px: "xl",
      name: "chevron-up",
      fontFamily: "Feather",
      position: "absolute",
      right: 0,
      color: "white"
    })
  });
  const {
    suffix,
    activeSuffix,
    prefix,
    activePrefix,
    active
  } = props;
  return /*#__PURE__*/React.createElement(Button, _extends({}, props, {
    prefix: active ? activePrefix ?? prefix : prefix,
    suffix: active ? activeSuffix ?? suffix : suffix
  }));
}; // CollapseHeader.defaultProps = {
//   block: true,
//   roundedBottom: 'none',
//   bg: 'blue600',
//   color: 'white',
//   py: 'xl',
//   px: 'none',
//   justifyContent: 'flex-start',
//   prefix: <Div px="lg" />,
//   suffix: (
//     <Icon
//       px="xl"
//       name="chevron-down"
//       fontFamily="Feather"
//       position="absolute"
//       right={0}
//       color="white"
//     />
//   ),
//   activeSuffix: (
//     <Icon
//       px="xl"
//       name="chevron-up"
//       fontFamily="Feather"
//       position="absolute"
//       right={0}
//       color="white"
//     />
//   ),
// };


export { CollapseHeader };
