// react-native-magnus/lib/commonjs/ui/checkbox/checkbox.component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Checkbox = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _checkbox = require("./checkbox.style");

var _utilities = require("../../utilities");

var _checkbox2 = require("./checkbox.service");

var _group = require("./group.component");

var _useDefaultProps = require("../../utilities/useDefaultProps");

var _theme = require("../../theme");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Checkbox = incomingProps => {
  const props = (0, _useDefaultProps.useDefaultProps)('Checkbox', incomingProps, {
    defaultChecked: false,
    activeColor: 'blue600',
    inactiveColor: 'gray500',
    highlightBg: 'gray300',
    bg: 'transparent',
    p: 'none',
    color: 'white',
    rounded: 'md',
    loading: false,
    disabled: false,
    loaderSize: '2xl',
    loaderColor: 'blue600',
    block: false,
    position: 'relative',
    shadowColor: 'gray800',
    shadow: 0,
    fontSize: '5xl',
    borderless: true,
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'flex-start',
    onPress: () => {},
    flexDir: 'row'
  });
  const {
    m,
    mt,
    mr,
    mb,
    ml,
    ms,
    p,
    pr,
    pt,
    pb,
    pl,
    h,
    w,
    bg,
    minW,
    minH,
    suffix,
    style,
    fontSize,
    fontWeight,
    prefix,
    rounded,
    roundedTop,
    roundedRight,
    roundedBottom,
    roundedLeft,
    color: colorProp,
    loading,
    disabled,
    loaderColor,
    loaderSize,
    children,
    borderColor,
    borderBottomColor,
    borderLeftColor,
    borderTopColor,
    borderRightColor,
    borderWidth,
    borderLeftWidth,
    borderRightWidth,
    borderBottomWidth,
    borderTopWidth,
    borderEndWidth,
    shadow,
    borderless,
    shadowColor,
    block,
    alignSelf,
    activeIcon,
    inactiveIcon,
    onChange,
    value,
    activeColor,
    inactiveColor,
    defaultChecked,
    checked: checkedProp,
    onChecked,
    onPress: onPressProp,
    ...rest
  } = props;
  const {
    theme
  } = (0, _theme.useTheme)();
  const [checked, setChecked] = (0, React.useState)(('checked' in props ? checkedProp : defaultChecked) ?? false);
  const [focussed, setFocussed] = (0, React.useState)(false);
  const computedStyle = (0, _checkbox.getStyle)(theme, props, {
    focussed
  });
  (0, React.useEffect)(() => {
    if ('checked' in props) {
      setChecked(props.checked ?? false);
    }
  }, [props]);
  /**
   * on press checkbox
   */

  const onPress = event => {
    if (disabled) {
      return;
    }

    setChecked(!checked);

    if ((0, _utilities.isFunction)(onChecked)) {
      onChecked(!checked);
    }

    if ((0, _utilities.isFunction)(onPressProp)) {
      onPressProp(event);
    } // if onChange prop is a valid function, call it


    if ((0, _utilities.isFunction)(onChange)) {
      onChange(value);
    }
  };
  /**
   * sets focussed to true
   *
   * @param event
   */


  const onPressIn = () => {
    setFocussed(true);
  };
  /**
   * sets focussed to true
   *
   * @param event
   */


  const onPressOut = () => {
    setFocussed(false);
  };

  const renderChildren = () => {
    if ((0, _utilities.isFunction)(children)) {
      return children({
        focussed,
        disabled: disabled ?? false,
        checked,
        loading
      });
    }

    return /*#__PURE__*/React.createElement(React.Fragment, null, prefix, /*#__PURE__*/React.createElement(_reactNative.View, null, /*#__PURE__*/React.createElement(_reactNative.View, {
      style: computedStyle.highlightContainer
    }), /*#__PURE__*/React.createElement(_reactNative.View, {
      style: computedStyle.icon
    }, icon)), children, suffix);
  };

  const icon = (0, _checkbox2.getIcon)(theme, props, checked);
  return /*#__PURE__*/React.createElement(_reactNative.Pressable, _extends({}, rest, {
    style: computedStyle.button,
    onPress: disabled || loading ? undefined : onPress,
    onPressIn: disabled || loading ? undefined : onPressIn,
    onPressOut: disabled || loading ? undefined : onPressOut
  }), /*#__PURE__*/React.createElement(_reactNative.View, {
    style: computedStyle.container
  }, renderChildren()));
}; // Checkbox.defaultProps = {
//   defaultChecked: false,
//   activeColor: 'blue600',
//   inactiveColor: 'gray500',
//   highlightBg: 'gray300',
//   bg: 'transparent',
//   p: 'none',
//   color: 'white',
//   rounded: 'md',
//   loading: false,
//   disabled: false,
//   loaderSize: '2xl',
//   loaderColor: 'blue600',
//   block: false,
//   position: 'relative',
//   shadowColor: 'gray800',
//   shadow: 0,
//   fontSize: '5xl',
//   borderless: true,
//   alignItems: 'center',
//   justifyContent: 'center',
//   alignSelf: 'flex-start',
//   onPress: () => {},
//   flexDir: 'row',
// };


exports.Checkbox = Checkbox;
Checkbox.Group = _group.CheckboxGroup;
