// react-native-magnus/lib/commonjs/ui/animated/animated.component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Animated = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var Animatable = _interopRequireWildcard(require("react-native-animatable"));

var _theme = require("../../theme");

var _animated = require("./animated.style");

var _animated2 = require("./animated.service");

var _useDefaultProps = require("../../utilities/useDefaultProps");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

(0, _animated2.registerAnimations)();

const Animated = incomingProps => {
  const props = (0, _useDefaultProps.useDefaultProps)('Animated', incomingProps, {
    bg: 'transparent',
    flexWrap: 'nowrap',
    rounded: 'none',
    shadow: 'none',
    shadowColor: 'gray900',
    position: 'relative',
    bgMode: 'cover',
    animation: 'fromTop',
    duration: 150,
    delay: 0
  });
  const {
    h,
    w,
    m,
    mt,
    mr,
    mb,
    ml,
    p,
    pr,
    pt,
    pb,
    pl,
    minH,
    minW,
    position,
    style,
    flexDir,
    rounded,
    roundedTop,
    roundedRight,
    roundedBottom,
    roundedLeft,
    children,
    bgImg,
    duration,
    bgMode,
    animation,
    alignItems,
    justifyContent,
    borderColor,
    borderBottomColor,
    borderLeftColor,
    borderTopColor,
    borderRightColor,
    borderWidth,
    borderLeftWidth,
    borderRightWidth,
    borderBottomWidth,
    borderTopWidth,
    flexWrap,
    shadow,
    shadowColor,
    delay,
    ...rest
  } = props;
  const {
    theme
  } = (0, _theme.useTheme)();
  const computedStyle = (0, _animated.getStyle)(theme, props);

  if (bgImg) {
    return /*#__PURE__*/React.createElement(Animatable.View, {
      animation: `magnus-${animation}`,
      duration: duration,
      easing: "ease-in-out",
      iterationCount: 1
    }, /*#__PURE__*/React.createElement(_reactNative.ImageBackground, _extends({
      source: bgImg,
      style: computedStyle.div,
      resizeMode: props.bgMode,
      imageStyle: computedStyle.image
    }, rest), children));
  }

  return /*#__PURE__*/React.createElement(Animatable.View, {
    animation: `magnus-${animation}`,
    duration: duration,
    easing: "ease-in-out",
    iterationCount: 1,
    delay: delay
  }, /*#__PURE__*/React.createElement(_reactNative.View, _extends({
    style: { ...computedStyle.div
    }
  }, rest), children));
}; // Animated.defaultProps = {
//   bg: 'transparent',
//   flexWrap: 'nowrap',
//   rounded: 'none',
//   shadow: 'none',
//   shadowColor: 'gray900',
//   position: 'relative',
//   bgMode: 'cover',
//   animation: 'fromTop',
//   duration: 150,
//   delay: 0,
// };


exports.Animated = Animated;
