// react-native-magnus/lib/commonjs/ui/snackbar/snackbar.component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Snackbar = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _toast = require("./toast.component");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class Snackbar extends _react.Component {
  constructor(props) {
    super(props);

    _defineProperty(this, "show", (message, config) => {
      let id = (config === null || config === void 0 ? void 0 : config.id) || Math.random().toString();

      const onClose = () => this.hide(id);

      requestAnimationFrame(() => {
        this.setState({
          toasts: this.state.toasts.filter(t => t.id !== id)
        });
        this.setState({
          toasts: [{
            id,
            message,
            onClose,
            ...config
          }, ...this.state.toasts]
        });
      });
      return id;
    });

    _defineProperty(this, "update", (id, message, config) => {
      this.setState({
        toasts: this.state.toasts.map(toast => toast.id === id ? { ...toast,
          message,
          ...config
        } : toast)
      });
    });

    _defineProperty(this, "hide", id => {
      this.setState({
        toasts: this.state.toasts.filter(t => t.id !== id)
      });
    });

    this.state = {
      toasts: []
    };
  }

  render() {
    const {
      toasts
    } = this.state;
    const {
      placement,
      offset
    } = this.props;
    let style = {
      bottom: placement === 'bottom' ? offset : undefined,
      top: placement === 'top' ? offset : undefined,
      justifyContent: placement === 'bottom' ? 'flex-end' : 'flex-start',
      flexDirection: placement === 'bottom' ? 'column' : 'column-reverse'
    };
    return /*#__PURE__*/_react.default.createElement(_reactNative.KeyboardAvoidingView, {
      behavior: _reactNative.Platform.OS === 'ios' ? 'position' : undefined,
      style: [styles.container, style],
      pointerEvents: "box-none"
    }, toasts.map(toast => {
      const {
        message,
        id,
        ...rest
      } = toast;
      return /*#__PURE__*/_react.default.createElement(_toast.Toast, _extends({
        key: id
      }, this.props, rest), message);
    }));
  }

}

exports.Snackbar = Snackbar;

_defineProperty(Snackbar, "defaultProps", {
  placement: 'bottom',
  offset: 0
});

const styles = _reactNative.StyleSheet.create({
  container: {
    position: 'absolute',
    width: '100%'
  }
});
