// react-native-magnus/lib/module/ui/select/select.component.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import RNModal from 'react-native-modal';
import { SafeAreaView, FlatList } from 'react-native';
import { useState, useImperativeHandle, useEffect, useMemo } from 'react';
import { getStyle } from './select.style';
import { Div } from '../div/div.component';
import { useTheme } from '../../theme';
import { Text } from '../text/text.component';
import { Input } from '../input/input.component';
import { Icon } from '../icon/icon.component';
import { SelectOption } from './select.option.component';
import { Button } from '../button/button.component';
import { useDefaultProps } from '../../utilities/useDefaultProps';
const Select = /*#__PURE__*/React.forwardRef((incomingProps, ref) => {
  const props = useDefaultProps('Select', incomingProps, {
    bg: 'white',
    rounded: 'none',
    flexDir: 'column',
    isVisible: false,
    renderNoResultsView: searchTerm => /*#__PURE__*/React.createElement(Div, {
      flex: 1,
      px: "2xl",
      py: "xl"
    }, /*#__PURE__*/React.createElement(Text, {
      fontSize: "lg"
    }, "No results found for \"", searchTerm, "\"")),
    keyExtractor: (_, index) => `${index}`
  });
  const {
    value,
    title,
    message,
    footer,
    data,
    multiple,
    renderItem,
    keyExtractor,
    renderNoResultsView,
    renderSubmitButton,
    renderSearchInput,
    searchableProps,
    onSelect: onSelectProp
  } = props;
  const {
    theme
  } = useTheme();
  const [visible, setVisible] = useState(props.isVisible || false);
  const [selectedValue, setSelectedValue] = useState(value);
  const [searchTerm, setSearchTerm] = useState('');
  const isSearchable = useMemo(() => !!(searchableProps !== null && searchableProps !== void 0 && searchableProps.length), [searchableProps]);
  const computedStyle = getStyle(theme, props);

  const resolveMultiLevelAccess = (obj, key) => {
    return key.split('.').reduce((cur, keySection) => {
      if (cur === null || cur === undefined) {
        return;
      }

      if (cur[keySection] === null || cur[keySection] === undefined) {
        console.warn(`Property "${key}" does not exists! `);
        return;
      }

      return cur[keySection];
    }, obj);
  };

  const filteredData = useMemo(() => {
    if (!searchableProps || Array.isArray(searchableProps) && !searchableProps.length) {
      return data;
    }

    return data.filter(item => {
      const lowSearch = searchTerm.toLowerCase();

      if (!Array.isArray(searchableProps)) {
        return String(item).toLowerCase().includes(lowSearch);
      }

      return searchableProps.some(key => String(resolveMultiLevelAccess(item, key)).toLowerCase().includes(lowSearch));
    });
  }, [searchableProps, searchTerm, data]);
  useEffect(() => {
    if ('isVisible' in props) {
      setVisible(props.isVisible || visible);
    }
  }, [props, visible]);
  useEffect(() => {
    if (visible) {
      clearSearchInput();
    }
  }, [visible]);
  /**
   * exposing functions to parent
   */

  useImperativeHandle(ref, () => ({
    close() {
      setVisible(false);
    },

    open() {
      setVisible(true);
    }

  }));
  /**
   * set values of select based if its multiple select
   * or single valued select
   *
   * @param value
   */

  const onSelect = value => {
    let finalValue;

    if (multiple) {
      const copy = selectedValue.slice();
      const index = selectedValue.indexOf(value);

      if (index !== -1) {
        copy.splice(index, 1);
      } else {
        copy.push(value);
      }

      setSelectedValue(copy);
      finalValue = copy;
    } else {
      setSelectedValue(value);
      setVisible(false);
      finalValue = value;
    }

    onSelectProp(finalValue);
  };
  /**
   * render title at top select modal
   */


  const renderTitle = () => {
    if (title) {
      return typeof title === 'string' ? /*#__PURE__*/React.createElement(Text, {
        px: "xl",
        fontSize: "lg",
        fontWeight: "bold",
        textTransform: "uppercase"
      }, title) : title;
    }

    return false;
  };
  /**
   * render message at top select modal
   */


  const renderMessage = () => {
    if (message) {
      return typeof message === 'string' ? /*#__PURE__*/React.createElement(Text, {
        px: "xl",
        fontSize: "lg"
      }, message) : message;
    }

    return false;
  };

  const clearSearchInput = () => setSearchTerm('');
  /**
   * render searchbar at top select modal
   */


  const renderSearchbar = () => {
    if (!isSearchable) {
      return;
    }

    const searchInputElement = renderSearchInput && renderSearchInput({
      clearText: clearSearchInput
    });
    const mandatoryProps = {
      value: searchTerm,
      onChangeText: text => setSearchTerm(text),
      autoComplete: 'off'
    };

    if (searchInputElement) {
      return /*#__PURE__*/React.cloneElement(searchInputElement, mandatoryProps);
    }

    return /*#__PURE__*/React.createElement(Input, _extends({
      mx: "xl",
      mt: "-md",
      fontSize: "lg",
      placeholder: "Search items",
      bg: "gray200",
      prefix: /*#__PURE__*/React.createElement(Icon, {
        mr: "lg",
        name: "search1",
        color: "gray700",
        fontSize: "3xl"
      }),
      suffix: searchTerm ? /*#__PURE__*/React.createElement(Button, {
        mx: "-md",
        p: "md",
        alignSelf: "center",
        rounded: "circle",
        bg: "gray400",
        onPress: clearSearchInput
      }, /*#__PURE__*/React.createElement(Icon, {
        name: "close",
        fontSize: "xs",
        color: "gray900"
      })) : null
    }, mandatoryProps));
  };

  const renderFooter = () => {
    if (footer) {
      return footer;
    } // if the component is single-valued and no footer is provided
    // don't render anything in footer


    if (!multiple) {
      return;
    }

    const submitButtonElement = renderSubmitButton && renderSubmitButton();
    const mandatoryProps = {
      onPress: () => setVisible(false)
    };

    if (submitButtonElement) {
      return /*#__PURE__*/React.cloneElement(submitButtonElement, mandatoryProps);
    }

    return /*#__PURE__*/React.createElement(Button, _extends({
      block: true,
      rounded: "none",
      bg: "green600"
    }, mandatoryProps), /*#__PURE__*/React.createElement(Text, {
      fontSize: "xl",
      py: "xs",
      color: "white"
    }, "Submit"));
  };

  const renderNoResultsFound = () => {
    renderNoResultsView && renderNoResultsView(searchTerm);
  };

  return /*#__PURE__*/React.createElement(RNModal, {
    backdropTransitionOutTiming: 0,
    animationIn: "slideInUp",
    isVisible: visible,
    backdropColor: "black",
    onBackdropPress: () => setVisible(false),
    onBackButtonPress: () => setVisible(false),
    hideModalContentWhileAnimating: true,
    style: {
      margin: 0,
      justifyContent: 'flex-end'
    }
  }, /*#__PURE__*/React.createElement(Div, {
    style: computedStyle.wrapper
  }, /*#__PURE__*/React.createElement(SafeAreaView, {
    style: computedStyle.container
  }, /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Div, null, /*#__PURE__*/React.createElement(Div, {
    py: "xl"
  }, renderTitle(), renderMessage()), /*#__PURE__*/React.createElement(Div, null, renderSearchbar())), filteredData.length > 0 ? /*#__PURE__*/React.createElement(Div, {
    flex: 1
  }, /*#__PURE__*/React.createElement(FlatList, {
    data: filteredData,
    keyExtractor: keyExtractor,
    renderItem: _ref => {
      let {
        item,
        index
      } = _ref;
      return /*#__PURE__*/React.cloneElement(renderItem(item, index), {
        onSelect,
        selectedValue
      });
    }
  })) : renderNoResultsFound(), renderFooter()))));
}); // Select.defaultProps = {
//   bg: 'white',
//   rounded: 'none',
//   flexDir: 'column',
//   isVisible: false,
//   // mb: 'xl',
//   // @ts-ignore
//   renderNoResultsView: (searchTerm) => (
//     <Div flex={1} px="2xl" py="xl">
//       <Text fontSize="lg">No results found for "{searchTerm}"</Text>
//     </Div>
//   ),
//   keyExtractor: (_, index) => `${index}`,
// };

Select.Option = SelectOption;
export { Select };
