// react-native-magnus/lib/commonjs/ui/scrolldiv/scrolldiv.component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ScrollDiv = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _scrolldiv = require("./scrolldiv.style");

var _theme = require("../../theme");

var _useDefaultProps = require("../../utilities/useDefaultProps");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const ScrollDiv = incomingProps => {
  const props = (0, _useDefaultProps.useDefaultProps)('ScrollDiv', incomingProps, {
    bg: 'transparent',
    flexDir: 'column',
    flexWrap: 'nowrap',
    rounded: 'none',
    shadow: 0,
    shadowColor: 'gray900',
    position: 'relative',
    bgMode: 'cover'
  });
  const {
    h,
    w,
    m,
    mt,
    mr,
    mb,
    ml,
    p,
    pr,
    pt,
    pb,
    pl,
    minH,
    minW,
    position,
    style,
    flexDir,
    rounded,
    roundedTop,
    roundedRight,
    roundedBottom,
    roundedLeft,
    children,
    bgImg,
    bgMode,
    alignItems,
    justifyContent,
    borderColor,
    borderBottomColor,
    borderLeftColor,
    borderTopColor,
    borderRightColor,
    borderWidth,
    borderLeftWidth,
    borderRightWidth,
    borderBottomWidth,
    borderTopWidth,
    flexWrap,
    shadow,
    shadowColor,
    ...rest
  } = props;
  const {
    theme
  } = (0, _theme.useTheme)();
  const computedStyle = (0, _scrolldiv.getStyle)(theme, props);

  if (bgImg) {
    return /*#__PURE__*/React.createElement(_reactNative.ImageBackground, _extends({
      source: bgImg,
      style: computedStyle.div,
      resizeMode: props.bgMode,
      imageStyle: computedStyle.image
    }, rest), children);
  }

  return /*#__PURE__*/React.createElement(_reactNative.ScrollView, _extends({
    style: computedStyle.div
  }, rest), children);
}; // ScrollDiv.defaultProps = {
//   bg: 'transparent',
//   flexDir: 'column',
//   flexWrap: 'nowrap',
//   rounded: 'none',
//   shadow: 0,
//   shadowColor: 'gray900',
//   position: 'relative',
//   bgMode: 'cover',
// };


exports.ScrollDiv = ScrollDiv;
