// react-native-magnus/lib/module/ui/carousel/carousel.component.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { useDefaultProps } from '../../utilities/useDefaultProps';
import { Div } from '../div/div.component';
import { ScrollDiv } from '../scrolldiv/scrolldiv.component';
import CarouselItem from './item.carousel';

const Carousel = incomingProps => {
  const props = useDefaultProps('Carousel', incomingProps, {
    showIndicators: true
  });
  const {
    children,
    renderIndicators,
    showIndicators
  } = props;
  const [selectedPage, setSelectedPage] = React.useState(1);
  const [totalPages, setTotalPages] = React.useState(1);
  const [totalContentWidth, setTotalContentWidth] = React.useState(0);
  const items = React.useMemo(() => {
    return React.Children.map(children, child => {
      if (! /*#__PURE__*/React.isValidElement(child)) {
        return;
      }

      if (child.type === CarouselItem) {
        return child;
      }

      console.warn('Only children of type Carousel.Item are allowed here. Any other child components will be ignored.');
      return null;
    }) ?? [];
  }, [children]);

  const init = width => {
    // initialize width
    setTotalContentWidth(width); // initialize total pages

    const totalItems = items.length;
    setTotalPages(totalItems);
  };

  const getPage = event => {
    const widthOffset = event.nativeEvent.contentOffset.x;
    const widthOfEachItem = totalContentWidth / totalPages; // const pageWidth = widthOfEachItem * itemsPerPage;

    for (let page = 1; page <= totalPages; page++) {
      if (widthOffset + 1 < widthOfEachItem * page) {
        return page;
      }

      if (page === totalPages) {
        return page;
      }
    }

    return 0;
  };

  const internal_renderIndicators = () => {
    if (renderIndicators) {
      return renderIndicators({
        selectedPage,
        totalPages
      });
    }

    return /*#__PURE__*/React.createElement(Div, {
      flexDir: "row",
      justifyContent: "center",
      my: "lg"
    }, Array(totalPages).fill(0).map((_, index) => {
      return /*#__PURE__*/React.createElement(Div, {
        w: 10,
        h: 10,
        rounded: "circle",
        bg: selectedPage === index + 1 ? 'blue600' : 'gray400',
        mx: "xs",
        key: index
      });
    }));
  };

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ScrollDiv, _extends({}, props, {
    horizontal: true,
    contentContainerStyle: {
      width: `${100 * totalPages}%`,
      flex: 1
    },
    showsHorizontalScrollIndicator: false,
    onContentSizeChange: (w, _h) => init(w),
    onScroll: data => {
      setTotalContentWidth(data.nativeEvent.contentSize.width);
      setSelectedPage(getPage(data));
    },
    scrollEventThrottle: 200,
    pagingEnabled: true,
    decelerationRate: "fast"
  }), items === null || items === void 0 ? void 0 : items.map((child, index) => /*#__PURE__*/React.createElement(Div, {
    flex: 1,
    key: index
  }, child))), showIndicators && internal_renderIndicators());
};

Carousel.Item = CarouselItem;
export { Carousel };
