// react-native-magnus/lib/module/ui/tag/tag.component.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from 'react';
import { useDefaultProps } from '../../utilities/useDefaultProps';
import { Button } from '../button/button.component';

const Tag = incomingProps => {
  const props = useDefaultProps('Tag', incomingProps, {
    px: 'lg',
    py: 'sm',
    bg: 'gray400',
    color: 'black',
    fontSize: 'lg',
    textAlign: 'auto',
    textTransform: 'none',
    rounded: 'md',
    borderWidth: 0,
    borderColor: 'transparent',
    onPress: () => {}
  });
  const {
    underlayColor,
    ...rest
  } = props;
  return /*#__PURE__*/React.createElement(Button, _extends({}, rest, {
    underlayColor: underlayColor ?? props.bg
  }));
};

export { Tag };
