// react-native-magnus/lib/commonjs/ui/select/select.option.component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectOption = void 0;

var React = _interopRequireWildcard(require("react"));

var _div = require("../div/div.component");

var _icon = require("../icon/icon.component");

var _text = require("../text/text.component");

var _button = require("../button/button.component");

var _useDefaultProps = require("../../utilities/useDefaultProps");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const SelectOption = incomingProps => {
  const props = (0, _useDefaultProps.useDefaultProps)('SelectOption', incomingProps, {
    onSelect: () => {},
    rounded: 0,
    bg: 'white',
    px: 'xl',
    py: 'lg',
    color: 'black',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    center: false,
    prefix: /*#__PURE__*/React.createElement(_icon.Icon, {
      name: "ios-checkmark-circle",
      fontFamily: "Ionicons",
      fontSize: "2xl",
      color: "green600",
      mr: "md"
    })
  });
  const {
    children,
    value,
    prefix,
    onPress: onPressProp,
    onSelect,
    selectedValue,
    ...rest
  } = props;
  const isSelected = Array.isArray(selectedValue) ? selectedValue.includes(value) : selectedValue === value;
  /**
   * on press select option
   *
   * @param e
   */

  const onPress = event => {
    if (onSelect) {
      onSelect(value);
    }

    if (onPressProp) {
      onPressProp(event);
    }
  };
  /**
   * render suffix
   */


  const renderPrefix = () => {
    if (isSelected) {
      return prefix;
    }

    return null;
  };

  const renderChildren = () => {
    if (typeof children === 'string') {
      return /*#__PURE__*/React.createElement(_text.Text, null, children);
    }

    return children;
  };

  return /*#__PURE__*/React.createElement(_button.Button, _extends({}, rest, {
    onPress: onPress,
    block: true,
    alignItems: "center"
  }), prefix && /*#__PURE__*/React.createElement(_div.Div, {
    minW: "8%"
  }, renderPrefix()), /*#__PURE__*/React.createElement(_div.Div, {
    flex: 1,
    bg: "transparent"
  }, renderChildren()));
}; // Option.defaultProps = {
//   onSelect: () => {},
//   rounded: 0,
//   bg: 'white',
//   p: 0,
//   color: 'black',
//   alignItems: 'flex-start',
//   justifyContent: 'flex-start',
//   center: false,
// };


exports.SelectOption = SelectOption;
