// react-native-magnus/lib/module/types.js
export const borderProps = ['borderColor', 'borderTopColor', 'borderRightColor', 'borderBottomColor', 'borderLeftColor', 'borderWidth', 'borderTopWidth', 'borderRightWidth', 'borderLeftWidth', 'borderBottomWidth', 'borderStartWidth', 'borderEndWidth', 'borderStyle'];
export const spacingProps = ['m', 'mt', 'mr', 'mb', 'ml', 'mx', 'my', 'ms', 'p', 'pt', 'pr', 'pb', 'pl', 'px', 'py', 'ps'];
export const roundedProps = ['rounded', 'roundedTopLeft', 'roundedTopRight', 'roundedBottomLeft', 'roundedBottomRight', 'roundedTop', 'roundedLeft', 'roundedRight', 'roundedBottom'];
export const shadowProps = ['shadow', 'shadowColor'];
export const dimensionProps = ['minH', 'minW', 'maxH', 'maxW', 'h', 'w'];
export const flexProps = ['row', 'flex', 'flexDir', 'flexWrap', 'justifyContent', 'alignSelf', 'alignItems'];
export const positionProps = ['position', 'top', 'right', 'bottom', 'left'];
export const backgroundProps = ['bg', 'bgImg', 'bgMode'];
export const textProps = ['color', 'fontSize', 'textDecorLine', 'textDecorStyle', 'fontStyle', 'textDecorColor', 'fontWeight', 'fontFamily', 'lineHeight', 'textAlign', 'textTransform', 'letterSpacing', 'textAlignVertical'];
export const opacityProps = ['opacity'];
export const overflowProps = ['overflow'];
export const zIndexProps = ['zIndex'];
export const loadingProps = ['loading', 'loaderSize', 'loaderColor'];
export const preffixSuffixProps = ['preffix', 'suffix'];
export const inputProps = ['focusBorderColor'];
export const disabledProps = ['disabled'];
export const buttonProps = ['underlayColor', 'block', 'borderless', 'rippleColor', 'ripple'];
export const overlayProps = ['overlayColor', 'overlayOpacity'];
export const variantProps = ['variant'];
