// react-native-magnus/lib/commonjs/ui/dropdown/dropdown.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getStyle = void 0;

var _reactNative = require("react-native");

var _theme = require("../../theme/theme.service");

/**
 * computed style
 *
 * @param theme
 * @param props
 */
const getStyle = (theme, props) => {
  const computedStyle = {};
  computedStyle.wrapper = {
    height: props.h,
    width: props.w,
    minHeight: props.minH,
    minWidth: props.minW,
    alignSelf: 'center',
    overflow: props.overflow,
    backgroundColor: (0, _theme.getThemeColor)(theme.colors, props.bg),
    ...(0, _theme.createBorderWidthStyles)(props),
    ...(0, _theme.createBorderColorStyles)(props, theme.colors),
    ...(0, _theme.createBorderRadiusStyles)(props, theme.borderRadius)
  };
  computedStyle.indicator = {
    alignSelf: 'center',
    marginVertical: 10
  };
  computedStyle.container = { ...(0, _theme.createSpacingStyles)(props, theme.spacing)
  };
  computedStyle.options = {
    flexWrap: props.flexWrap,
    flexDirection: props.flexDir,
    alignItems: props.alignItems,
    justifyContent: props.justifyContent
  }; // merging style props to computed style

  if (props.style) {
    computedStyle.container = { ...computedStyle.container,
      // @ts-ignore
      ...props.style
    };
  }

  return _reactNative.StyleSheet.create(computedStyle);
};

exports.getStyle = getStyle;
