// react-native-magnus/lib/commonjs/ui/fab/fab.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getStyle = void 0;

var _reactNative = require("react-native");

var _theme = require("../../theme/theme.service");

/**
 * computed style
 *
 * @param theme
 * @param props
 */
const getStyle = (theme, props) => {
  const computedStyle = {};
  computedStyle.button = {
    bottom: 0,
    right: 0,
    overflow: _reactNative.Platform.OS === 'ios' ? 'visible' : 'hidden',
    zIndex: 2,
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 5,
    position: props.position
  };
  computedStyle.actions = {
    position: 'absolute',
    zIndex: 10
  };

  if (props.shadow) {
    computedStyle.button = { ...computedStyle.button,
      ...(theme.shadow && theme.shadow[props.shadow]),
      shadowColor: (0, _theme.getThemeColor)(theme.colors, props.shadowColor)
    };
  }

  computedStyle.overlay = {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
    elevation: 0,
    zIndex: 0
  };
  return _reactNative.StyleSheet.create(computedStyle);
};

exports.getStyle = getStyle;
