// react-native-magnus/lib/commonjs/ui/image/image.style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getStyle = void 0;

var _reactNative = require("react-native");

var _theme = require("../../theme/theme.service");

/**
 * computed style
 *
 * @param theme
 * @param props
 */
const getStyle = (theme, props) => {
  const computedStyle = {};
  computedStyle.image = {
    flex: props.flex,
    height: props.h,
    width: props.w,
    minHeight: props.minH,
    minWidth: props.minH,
    maxWidth: props.maxW,
    maxHeight: props.maxH,
    position: props.position,
    zIndex: props.zIndex,
    alignSelf: props.alignSelf,
    ...(0, _theme.createPositionStyle)(props),
    ...(0, _theme.createBorderWidthStyles)(props),
    ...(0, _theme.createShadowStyles)(props, theme),
    ...(0, _theme.createSpacingStyles)(props, theme.spacing),
    ...(0, _theme.createBorderColorStyles)(props, theme.colors),
    ...(0, _theme.createBorderRadiusStyles)(props, theme.borderRadius),
    backgroundColor: (0, _theme.getThemeColor)(theme.colors, props.bg)
  }; // merging style props to computed style

  if (props.style) {
    computedStyle.image = { ...computedStyle.image,
      // @ts-ignore
      ...props.style
    };
  }

  return _reactNative.StyleSheet.create(computedStyle);
};

exports.getStyle = getStyle;
